lucet-wasi/src/memory.rs

```diff
@@ -357,3 +357,72 @@ dec_enc_scalar!(
     enc_whence,
     enc_whence_byref
 );
+
+dec_enc_scalar!(
+    __wasi_subclockflags_t,
+    dec_subclockflags,
+    dec_subclockflags_byref,
+    enc_subclockflags,
+    enc_subclockflags_byref
+);
+
+dec_enc_scalar!(
+    __wasi_eventrwflags_t,
+    dec_eventrwflags,
+    dec_eventrwflags_byref,
+    enc_eventrwflags,
+    enc_eventrwflags_byref
+);
+
+dec_enc_scalar!(
+    __wasi_eventtype_t,
+    dec_eventtype,
+    dec_eventtype_byref,
+    enc_eventtype,
+    enc_eventtype_byref
+);
+
+pub fn dec_subscription(
+    subscription: &wasm32::__wasi_subscription_t,
+) -> Result<host::__wasi_subscription_t, host::__wasi_errno_t> {
+    let userdata = subscription.userdata;
+    let type_ = dec_eventtype(subscription.type_);
+    let u_orig = subscription.__bindgen_anon_1;
+    let u = match type_ {
+        wasm32::__WASI_EVENTTYPE_CLOCK => host::__wasi_subscription_t___wasi_subscription_u {
+            clock: unsafe {
+                host::__wasi_subscription_t___wasi_subscription_u___wasi_subscription_u_clock_t {
+                    identifier: u_orig.clock.identifier,
```

Comment:
another `dec_userdata`

Reply:
`userdata` is an opaque pointer passed back and forth by the wasm module, and that only makes sense it its own context. It's something that the syscall should never ever touch. Is it worth decoding/encoding it rather than assume that it will always be in wasm representation?

Reply:
we can't be sure of that without overriding what bindgen produces for the host side, unfortunately

Reply:
done.